src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -190,9 +190,9 @@ end
         @yield timeout(prot.sim, prot.local_busy_time_post)
 
         # tag local node a with EntanglementCounterpart remote_node_idx_b remote_slot_idx_b
-        tag!(a, EntanglementCounterpart, prot.nodeB, b.idx)
+        tag!(a, EntanglementCounterpart, prot.nodeB, b.idx; tag_time = now(prot.sim), id = guid())
```

Comment:
I do not think this is a viable solution. We can not ask the user to add a `guid` manually everywhere. This should be happening automatically behind the scenes.

Reply:
Same with `tag_time`. This and `id` maybe can optionally be overridden, but by default they should happen automatically.

One option for time is to do `now(get_time_tracker(a))` where `a` has internally a reference to the time tracker.